src/Behat/Config/Suite.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Behat\Config;
+
+final class Suite
+{
+    public function __construct(
+        private string $name,
+        private array $settings = [],
+        private string $profile = 'default',
+    ) {
+    }
+
+    public function withContexts(string ...$contexts): self
```

Comment:
The contexts can also be arrays in case you want to pass any parameters to the contexts, for example what we would previously see in a yaml config file:

```
default:
    suites:
        default:
            contexts:
                - MyAwesomeContext:
                    - http://localhost:8080
                    - /var/tmp
```

Reply:
array of string?

Reply:
In this case we would have:

```
$context = [
    'MyAwesomeContext' => [
        'http://localhost:8080',
        '/var/tmp',
    ]
]

Reply:
For the PHP config, I would suggest using a method `->addContext(class-string<Context> $context, array<mixed> $arguments = []): static` instead (of course using valid PHP types and moving the more precise types to phpdoc), and handling the representation of how registering a context with argument looks like inside that method instead of exposing it to the outside. The fact that registering arguments involves an array with a single argument using the class name as key and arguments as values is not something we can properly describe to static analyzers AFAICT.

Reply:
TBH, I like the explicit `->addContext(string $context, array $constructorArgs)` both because it hides our implementation detail of how that array is structured, and because it makes more obvious to the user that this is an available feature. I think if you're passing arguments, your config is already quite verbose and the repeated `addContext` calls are less significant.

However, I would **also** keep a simple `->withContexts(string ...$contexts)` for people who just want to provide a list of names with no extra detail.

Reply:
@acoulton that sounds like a nice compromise. 👍

Reply:
indeed. With an extension resolving contexts in a different way, it could be any string.

Reply:
Yes, this is from the friends of behat symfony extension.